Docker runtime: panic running `ig` with a `dead` container

## Description

I am seeing a panic when running `ig` with a docker container in [dead](https://github.com/docker/cli/issues/502#issuecomment-330361748) state:

```
→ sudo -E ig run trace_dns -r docker
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/docker.DockerContainerToContainerData(...)
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/docker/docker.go:283
github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/docker.(*DockerClient).GetContainers(0x0?)
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/docker/docker.go:133 +0x1a5
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager.(*localManager).initCollections.WithMultipleContainerRuntimesEnrichment.WithContainerRuntimeEnrichment.func13(0xc000442000)
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection/options.go:211 +0x371
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager.(*localManager).initCollections.WithMultipleContainerRuntimesEnrichment.func5(0xc000442000)
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection/options.go:158 +0x53
github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection.(*ContainerCollection).Initialize(0xc000442000, {0xc0003d9940, 0x8, 0x0?})
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection/container-collection.go:107 +0x9b
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager.(*localManager).initCollections(0xc00017ec00, {0xc001018250, 0x1, 0x10?}, {0x0, 0x0}, 0x0)
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager/localmanager.go:291 +0x806
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager.(*localManager).Init(0xc00017ec00, 0xc00017f1a0)
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager/localmanager.go:235 +0x9f0
github.com/inspektor-gadget/inspektor-gadget/cmd/common.NewRunCommand.func1(0xc00047c608, {0xc000418900, 0x3, 0x3})
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/cmd/common/oci.go:181 +0x706
github.com/spf13/cobra.(*Command).execute(0xc00047c608, {0xc000418900, 0x3, 0x3})
	/home/qasim/.gvm/pkgsets/go1.24.1/global/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1000 +0xa13
github.com/spf13/cobra.(*Command).ExecuteC(0xc00027cf08)
	/home/qasim/.gvm/pkgsets/go1.24.1/global/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x40c
github.com/spf13/cobra.(*Command).Execute(...)
	/home/qasim/.gvm/pkgsets/go1.24.1/global/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071
main.main()
	/home/qasim/work/cloud/github.com/inspektor-gadget/inspektor-gadget/cmd/ig/main.go:121 +0x4e9
```

## Impact

Not able to use `ig`

## Expected behavior

`ig` should gracefully handle `dead` container

## Additional information

It seems the issue is because of a `dead` container not having a [name](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-utils/docker/docker.go#L283).

```
→ docker ps -a --filter status=dead
CONTAINER ID   IMAGE     COMMAND   CREATED        STATUS    PORTS     NAMES
36d852a2a5d6   busybox   "sh"      2 months ago   Dead
```

--- a/pkg/container-utils/docker/docker.go
+++ b/pkg/container-utils/docker/docker.go
@@ -278,9 +278,13 @@ func containerStatusStateToRuntimeClientState(containerState string) (runtimeCli
 
 func DockerContainerToContainerData(container *container.Summary) *runtimeclient.ContainerData {
 	imageDigest := ""
+	containerName := ""
+	if len(container.Names) > 0 {
+		containerName = container.Names[0]
+	}
 	return buildContainerData(
 		container.ID,
-		container.Names[0],
+		containerName,
 		container.Image,
 		imageDigest,
 		container.State,

